jitrebalance/jitrebalance.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+from pyln.client import Plugin, Millisatoshi
+from pprint import pprint
+from remote_pdb import RemotePdb
+from math import ceil
+import secrets
+import hashlib
+import binascii
+import time
+
+plugin = Plugin()
+pdb = RemotePdb('127.0.0.1', 4444)
+
+
+def get_circular_route(route, scid):
+    pass
+
+
+@plugin.async_hook("htlc_accepted")
```

Comment:
so as the hook is async how do you make sure that the HTLC is only attempted to forwarded if rebalancing was succesfull? In particular: see my comment in line 148. Wouldn't it make sense to have more than 1 rebalancing attempt?

Reply:
> so as the hook is async how do you make sure that the HTLC is only attempted to forwarded if rebalancing was succesfull?

The async trick is only Pylightning side, to be able to "return" a result to `lightningd` without having to actually return in the plugin logic (see https://github.com/lightningd/plugins/pull/66/files#diff-78896cd8fe0da553087891ebc6efda26R30-R39)

> Wouldn't it make sense to have more than 1 rebalancing attempt?

Agree, even if I think it's intended. We could check all potential routes as in rebalance and drain plugins, and we could reduce the `1.25` factor on the rebalanced amount on capacity failure.